### PR TITLE
Speed up clean up and batch selects.

### LINKED ISSUE #530: Diagnose intermittent deadlocks
Continuing discussion from #438 on intermittent deadlocks:

```
WordPress database error Deadlock found when trying to get lock; try restarting transaction for query UPDATE wp_actionscheduler_actions SET attempts = attempts+1, status='in-progress', last_attempt_gmt = '2020-04-20 17:00:02', last_attempt_local = '2020-04-20 18:00:02' WHERE action_id = 1877421 made by do_action('wp_ajax_as_async_request_queue_runner'), WP_Hook->do_action, WP_Hook->apply_filters, WP_Async_Request->maybe_handle, ActionScheduler_AsyncRequest_QueueRunner->handle, do_action('action_scheduler_run_queue'), WP_Hook->do_action, WP_Hook->apply_filters, ActionScheduler_QueueRunner->run, ActionScheduler_QueueRunner->do_batch, ActionScheduler_Abstract_QueueRunner->process_action, ActionScheduler_DBStore->log_execution, QM_DB->query
```
and
```
WordPress database error Deadlock found when trying to get lock; try restarting transaction for query UPDATE wp_actionscheduler_actions SET attempts = attempts+1, status='in-progress', last_attempt_gmt = '2020-04-20 17:02:08', last_attempt_local = '2020-04-20 18:02:08' WHERE action_id = 1877473 made by do_action_ref_array('action_scheduler_run_queue'), WP_Hook->do_action, WP_Hook->apply_filters, ActionScheduler_QueueRunner->run, ActionScheduler_QueueRunner->do_batch, ActionScheduler_Abstract_QueueRunner->process_action, ActionScheduler_DBStore->log_execution
```

### PR DESCRIPTION
**Pull request by peterfabian**
TL;DR is that SELECT queries on the actions table are by default ordering the result by date_scheduled_gmt, which was orders of magnitude slower than non-sorted query variant. Thus, I've updated the places where I think the order doesn't play any role, and where it potentially plays tiny role, I've changed it to sort by action_id, which is the primary key, and theoretically the data thus should already be ordered by it. 

After the fact I implemented this, I noticed that @leewillis77 already noticed this and added their fix in #651, but this hopefully provides a bigger jump in overall performance, since the cleanup runs on each `ActionScheduler_QueueRunner::run`.

For further details from the research and comparison, please see my comments on #530. 

Closes #530 together with #680.

Testing instructions are the same as for #680.

### CODE REVIEW

**Review by imcodingideas — APPROVED** [review]

**Review by jonathansadowski — APPROVED** [review]
Tested this along with the changes in #680. Changes make sense to me and there seems to be a marked improvement, similar to the one noted in the test plan for 680. Nice work! I'm going to approve this, but leave it unmerged until the review for 680 is completed. /cc @barryhughes
